code/src/test/kotlin/com/expediagroup/sdk/core/logging/masking/PatternBuildersTest.kt

```diff
@@ -0,0 +1,97 @@
+package com.expediagroup.sdk.core.logging.masking
+
+import com.ebay.ejmask.extenstion.builder.json.JsonFullValuePatternBuilder
+import com.ebay.ejmask.extenstion.builder.json.JsonRelativeFieldPatternBuilder
+import com.expediagroup.sdk.core.logging.common.Constant.OMITTED
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class PatternBuildersTest {
+    @Nested
+    inner class CustomJsonFullValuePatternBuilderTest {
+        @Test
+        fun `is child of JsonFullValuePatternBuilder`() {
+            val custom = CustomJsonFullValuePatternBuilder()
+
+            assertEquals(JsonFullValuePatternBuilder::class.java, custom::class.java.superclass)
+        }
+
+        @Test
+        fun `does not override isGroupable`() {
+            val original = JsonFullValuePatternBuilder()
+            val custom = CustomJsonFullValuePatternBuilder()
+
+            assertEquals(original.isGroupable, custom.isGroupable)
+        }
+
+        @Test
+        fun `buildReplacement contains omitted keyword`() {
+            val custom = CustomJsonFullValuePatternBuilder()
+
+            assertTrue(custom.buildReplacement(0, "field").contains(OMITTED))
+        }
+
+        @Test
+        fun `buildReplacement always generates proper replacement template`() {
+            val custom = CustomJsonFullValuePatternBuilder()
+
+            custom.buildReplacement(0, "field1").also {
+                assertEquals(CustomJsonFullValuePatternBuilder.REPLACEMENT_TEMPLATE, it)
+            }
+
+            custom.buildReplacement(0, "field1", "field2").also {
+                assertEquals(CustomJsonFullValuePatternBuilder.REPLACEMENT_TEMPLATE, it)
+            }
+
+            custom.buildReplacement(0, "field".repeat(1000)).also {
+                assertEquals(CustomJsonFullValuePatternBuilder.REPLACEMENT_TEMPLATE, it)
+            }
+        }
+    }
+
+    @Nested
+    inner class CustomJsonRelativeFieldPatternBuilderTest {
+        @Test
+        fun `is child of JsonRelativeFieldPatternBuilder`() {
+            val custom = CustomJsonRelativeFieldPatternBuilder()
+
+            assertEquals(JsonRelativeFieldPatternBuilder::class.java, custom::class.java.superclass)
+        }
+
+        @Test
+        fun `does not override isGroupable`() {
```

Comment:
Why do we have test for this specific case?

Reply:
Overriding the `isGroupable` can change the overall behavior of fields masking and regex generation. Since that is not our desired outcome, I added a test for it.

Reply:
I think we're testing too much low level detail here, unit tests should check the **output** of a "unit" given some **input**. So, if that unit gives the "expected" output in different scenarios and cases, we shouldn't care about what's changed internally (blackbox).